mongodbatlas/fw_data_source_mongodbatlas_alert_configurations.go

```diff
@@ -0,0 +1,225 @@
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	matlas "go.mongodb.org/atlas/mongodbatlas"
+)
+
+const alertConfigurationsDataSourceName = "alert_configurations"
+
+var _ datasource.DataSource = &AlertConfigurationDS{}
+var _ datasource.DataSourceWithConfigure = &AlertConfigurationDS{}
+
+type tfAlertConfigurationsDSModel struct {
+	ID          types.String                  `tfsdk:"id"`
+	ProjectID   types.String                  `tfsdk:"project_id"`
+	OutputType  []string                      `tfsdk:"output_type"`
+	ListOptions []tfListOptionsModel          `tfsdk:"list_options"`
+	Results     []tfAlertConfigurationDSModel `tfsdk:"results"`
+	TotalCount  types.Int64                   `tfsdk:"total_count"`
+}
+
+type tfListOptionsModel struct {
+	PageNum      types.Int64 `tfsdk:"page_num"`
+	ItemsPerPage types.Int64 `tfsdk:"items_per_page"`
+	IncludeCount types.Bool  `tfsdk:"include_count"`
+}
+
+func NewAlertConfigurationsDS() datasource.DataSource {
+	return &AlertConfigurationsDS{}
+}
+
+type AlertConfigurationsDS struct {
+	client *MongoDBClient
+}
+
+func (d *AlertConfigurationsDS) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = fmt.Sprintf("%s_%s", req.ProviderTypeName, alertConfigurationsDataSourceName)
+}
+
+func (d *AlertConfigurationsDS) Configure(ctx context.Context, req datasource.ConfigureRequest, resp *datasource.ConfigureResponse) {
+	client, err := ConfigureClientInResource(req.ProviderData)
+	if err != nil {
+		resp.Diagnostics.AddError(errorConfigureSummary, err.Error())
+		return
+	}
+	d.client = client
+}
+
+func (d *AlertConfigurationsDS) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Computed: true,
+			},
+			"project_id": schema.StringAttribute{
+				Required: true,
+			},
+			"total_count": schema.Int64Attribute{
+				Computed: true,
+			},
+			"output_type": schema.ListAttribute{
+				Optional:    true,
+				ElementType: types.StringType,
+				Validators: []validator.List{
+					listvalidator.ValueStringsAre(stringvalidator.OneOf("resource_hcl", "resource_import")),
+				},
+			},
+			"results": schema.ListNestedAttribute{
+				Computed: true,
+				NestedObject: schema.NestedAttributeObject{
+					Attributes: copyAndAdd(alertConfigDSSchemaAttributes,
+						"output",
+						schema.ListNestedAttribute{
+							Computed: true,
+							NestedObject: schema.NestedAttributeObject{
+								Attributes: map[string]schema.Attribute{
+									"type": schema.StringAttribute{
+										Computed: true,
+									},
+									"label": schema.StringAttribute{
+										Computed: true,
+									},
+									"value": schema.StringAttribute{
+										Computed: true,
+									},
+								},
+							},
+						}),
+				},
+			},
+		},
+		Blocks: map[string]schema.Block{
```

Comment:
[q] Same as above, why is `list_options` here a block?

Reply:
Similar to the case above, to maintain backward compatibility for users defining the list options attribute, example:
```
data "mongodbatlas_alert_configurations" "test" {
	project_id = mongodbatlas_project.test.id

	list_options {
		page_num = 0
	}
}
```